### PR TITLE
:memo: Update Mask Generation Task

### PR DESCRIPTION
**Pull request by rajaryan18**
Added about.md for Zero-shot-object-detection, text-to-3d and mask-generation. Also added use cases for them all.

### CODE REVIEW

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## About the Task
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:1]:
```suggestion
```
No need to add this header, it's already added in `summary` in `data.ts`

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+Mask Generation task involves generating masks/segments for objects in a given image. It could be zero shot or based on training data.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:3]:
```suggestion
Mask generation task involves generating masks for meaningful segments in a given image. It could be zero-shot or based on training data.
```

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+## Use Cases
+
+Mask Generation has promising use cases in Medical Imaging, Autonomous Vehicles, Robotics, etc.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:7]: In this section we expect to have multiple headers for each use case.

File: packages/tasks/src/tasks/zero-shot-object-detection/about.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+## About the Task
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/zero-shot-object-detection/about.md:1]: This page is already contributed in another PR, so if you feel like contributing it, you can open PR to that branch as that one is more extensive

File: packages/tasks/src/tasks/text-to-3d/about.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+## About the Task
+
+Text to 3D is a Generative task of generating a 3d image from a given text prompt. It is similar to generating a 2d image but involves an extended complexity of depth perception.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/text-to-3d/about.md:3]: You can add this in `summary` in `data.ts`

```suggestion
Text-to-3D is to generate a 3D image from a text prompt. It is similar to generating a 2D image but involves an extended complexity of depth perception.
```

File: packages/tasks/src/tasks/text-to-3d/about.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+## About the Task
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/text-to-3d/about.md:1]: No need to add, we add this in `summary`

File: packages/tasks/src/tasks/text-to-3d/about.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+
+Text to 3D is a Generative task of generating a 3d image from a given text prompt. It is similar to generating a 2d image but involves an extended complexity of depth perception.
+
+## Use Cases
+
+Text to 3d has various creative and experimental use cases like setting up virtual experiments and simulations.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/text-to-3d/about.md:7]: Same comment applies here

**Review by merveenoyan — COMMENTED** [review]
Thanks a lot for this PR! You can add your HF user handle to end of `about.md`.

Moreover, feel free to remove the empty files if you won't contribute content inside.

**Comment by rajaryan18** [comment]
@merveenoyan 
Hey, I have added the changes you suggested. Thank you so much for your time.

File: .vscode/settings.json [hunk 8]
```diff
@@ -2,7 +2,7 @@
 	"editor.defaultFormatter": "esbenp.prettier-vscode",
```
Comment (merveenoyan) [inline .vscode/settings.json:2]: Can you remove this file altogether?

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+## Use Cases
+
+### Medical Imaging
+
+Mask Generation is used in medical imaging in aiding in the segmentation and analysis of specific regions.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:5]:
```suggestion
Mask generation models are used in medical imaging to aid in segmenting and analyzing specific regions.
```

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+
+### Autonomous Vehicles
+
+Mask Generation is used to create segments and masks for obstacles and other objects in view.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:9]:
```suggestion
Mask generation models are used to create segments and masks for obstacles and other objects in view.
```

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+
+This page was written by [Raj Aryan](https://huggingface.co/thatrajaryan)
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:11]:
```suggestion
This page was made possible thanks to the efforts of [Raj Aryan](https://huggingface.co/thatrajaryan).
```

File: packages/tasks/src/tasks/text-to-3d/data.ts [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+const taskData: TaskDataCustom = {
+	datasets: [
+		{
+			description: "3D-Future is a dataset designed to generate 3D images from textual descriptions.",
+			id: "3D-Future",
+		}
+	],
+	demo: {
+		inputs: [
+
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/text-to-3d/data.ts:12]: can you remove these gaps in the empty lists?

File: packages/tasks/src/tasks/text-to-3d/data.ts [hunk 13]
```diff
@@ -0,0 +1,30 @@
+import type { TaskDataCustom } from "..";
+
+const taskData: TaskDataCustom = {
+	datasets: [
+		{
+			description: "3D-Future is a dataset designed to generate 3D images from textual descriptions.",
+			id: "3D-Future",
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/text-to-3d/data.ts:7]: This dataset isn't on Hugging Face Hub, perhaps you can create the dataset? would be great apparently there's none on Hub and you'd be the first person to add such dataset!

File: packages/tasks/src/tasks/text-to-3d/data.ts [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+		],
+	},
+	metrics: [
+	],
+	models: [
+	],
+	spaces: [
+		
+	],
+	summary:"Text-to-3D is to generate a 3D image from a text prompt. It is similar to generating a 2D image but involves an extended complexity of depth perception.",
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/text-to-3d/data.ts:25]:
```suggestion
	summary: "Text-to-3D is the task of generating a 3D image from a text prompt. It is similar to generating a 2D image with an extension of depth perception.",
```

**Review by merveenoyan — COMMENTED** [review]
Thanks a lot!

**Comment by rajaryan18** [comment]
@merveenoyan 
Hey, I have made the changes.
Changed .vscode/setting.json back to its original content (from this repository).
Removed the dataset 3D-Future from data.ts. Will find it and add to the hub and then maybe modify this file again to mention it.

File: .vscode/settings.json [hunk 15]
```diff
@@ -7,4 +7,4 @@
 	"[svelte]": {
 		"editor.defaultFormatter": "esbenp.prettier-vscode"
 	}
-}
+}
```
Comment (merveenoyan) [inline .vscode/settings.json:10]: there seems to be a change still (see diff), if you can revert this change as well we can merge

**Review by merveenoyan — APPROVED** [review]
Thank you!

**Comment by rajaryan18** [comment]
@merveenoyan 
Hi again, I have changed that file. If it still crashes the checks, I'll just open a new PR and ensure to not commit any changes to .vscode folder.
Thank you

**Comment by merveenoyan** [comment]
@rajaryan18 can you merge this PR into your branch? https://github.com/rajaryan18/huggingface.js/pull/1 this will lint your code so the tests pass

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 16]
```diff
@@ -1,7 +1,16 @@
...
-	datasets: [],
+	datasets: [
+		{
+			description: "Widely used benchmark dataset for multiple Vision tasks.",
+			id: "merve/coco2017",
+		},
+        {
+            description: "Medical Imaging dataset of the Human Brain for segmentation and mask generating tasks",
+            id: "rocky93/BraTS_segmentation"
+        }
```
Comment (pcuenca) [inline packages/tasks/src/tasks/mask-generation/data.ts:12]:
```suggestion
		{
			description: "Widely used benchmark dataset for multiple Vision tasks.",
			id: "merve/coco2017",
		},
                {
                    description: "Medical Imaging dataset of the Human Brain for segmentation and mask generating tasks",
                    id: "rocky93/BraTS_segmentation"
                }
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 17]
```diff
@@ -16,7 +25,13 @@ const taskData: TaskDataCustom = {
 			},
 		],
 	},
-	metrics: [],
+	metrics: [
+		{
+			description:
+				"IoU is used to measure the overlap between predicted mask and the ground truth mask.",
```
Comment (pcuenca) [inline packages/tasks/src/tasks/mask-generation/data.ts:31]:
```suggestion
			description: "IoU is used to measure the overlap between predicted mask and the ground truth mask.",
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 18]
```diff
@@ -52,4 +67,4 @@ const taskData: TaskDataCustom = {
 	youtubeId: "",
 };
 
-export default taskData;
+export default taskData;
```
Comment (pcuenca) [inline packages/tasks/src/tasks/mask-generation/data.ts:70]:
```suggestion
export default taskData;

```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 19]
```diff
@@ -1,7 +1,16 @@
...
+	datasets: [
+		{
+			description: "Widely used benchmark dataset for multiple Vision tasks.",
+			id: "merve/coco2017",
+		},
+                {
+                    description: "Medical Imaging dataset of the Human Brain for segmentation and mask generating tasks",
+                    id: "rocky93/BraTS_segmentation"
+                }
+	],
```
Comment (pcuenca) [inline packages/tasks/src/tasks/mask-generation/data.ts:13]:
```suggestion
	datasets: [
		{
			description: "Widely used benchmark dataset for multiple Vision tasks.",
			id: "merve/coco2017",
		},
		{
			 description: "Medical Imaging dataset of the Human Brain for segmentation and mask generating tasks",
			 id: "rocky93/BraTS_segmentation"
		},
	],
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 20]
```diff
@@ -16,7 +25,12 @@ const taskData: TaskDataCustom = {
 			},
 		],
 	},
-	metrics: [],
+	metrics: [
+		{
+			description: "IoU is used to measure the overlap between predicted mask and the ground truth mask.",
+			id: "Intersection over Union (IoU)",
+		}
```
Comment (pcuenca) [inline packages/tasks/src/tasks/mask-generation/data.ts:32]:
```suggestion
		},
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 21]
```diff
@@ -1,7 +1,16 @@
...
-	datasets: [],
+	datasets: [
+		{
+			description: "Widely used benchmark dataset for multiple Vision tasks.",
+			id: "merve/coco2017",
+		},
+		{
+			 description: "Medical Imaging dataset of the Human Brain for segmentation and mask generating tasks",
+			 id: "rocky93/BraTS_segmentation"
+		},
```
Comment (pcuenca) [inline packages/tasks/src/tasks/mask-generation/data.ts:12]:
```suggestion
	datasets: [
		{
			description: "Widely used benchmark dataset for multiple Vision tasks.",
			id: "merve/coco2017",
		},
		{
			description: "Medical Imaging dataset of the Human Brain for segmentation and mask generating tasks",
			id: "rocky93/BraTS_segmentation",
		},
```

**Comment by pcuenca** [comment]
I fixed the conflicts and brought this up to date, but the linter still fails for some reason cc @coyotte508 @merveenoyan

**Comment by pcuenca** [comment]
Thank you @coyotte508! Merging now, thanks @rajaryan18 for the contribution and sorry for the delay 🙌
